models/policy_document.go

```diff
@@ -0,0 +1,81 @@
+package models
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type PolicyDocument struct {
+	Version   string
+	Statement []statement
+}
+
+type statement struct {
+	Sid       string
+	Effect    string
+	Principal map[string]string
+	Action    []string
+}
+
+func NewPolicyDocument() PolicyDocument {
+	return PolicyDocument{
+		Version:   "2008-10-17",
+		Statement: []statement{},
+	}
+}
+
+func newStatement(accountID string) statement {
+	return statement{
+		Sid:    accountID,
+		Effect: "allow",
+		Principal: map[string]string{
+			"AWS": fmt.Sprintf("arn:aws:iam::%s:root", accountID),
+		},
+		Action: []string{
+			"ecr:GetDownloadUrlForLayer",
+			"ecr:BatchGetImage",
+			"ecr:BatchCheckLayerAvailability",
+		},
+	}
+}
+
+func (p *PolicyDocument) AddAWSAccountPrincipal(accountID string) {
+	permissionExists := false
+	for _, s := range p.Statement {
+		if s.Sid == accountID {
+			permissionExists = true
+			break
+		}
+	}
+
+	if permissionExists {
+		return
+	}
+
+	p.Statement = append(p.Statement, newStatement(accountID))
+}
+
+func (p *PolicyDocument) RemoveAWSAccountPrincipal(accountID string) {
+	temp := []statement{}
+	for _, s := range p.Statement {
+		if s.Sid != accountID {
+			temp = append(temp, s)
+		}
+	}
+
+	p.Statement = temp
+}
+
+func (p *PolicyDocument) RenderPolicyText() string {
```

Comment:
hmm - the error swallowing is a little concerning here. Maybe we remove this helper function and allow the caller to properly handle errors?

Reply:
yea I wasn't quite sure what to do here either. In theory though this error should never be triggered as we're not trying to serialize a non-serializable type.